resources/providers/azurelib/inventory/provider.go

```diff
@@ -124,8 +126,8 @@ func (p *ProviderInitializer) getSubscriptionIds(ctx context.Context, azureConfi
 			return nil, err
 		}
 		for _, subscription := range page.Value {
-			if subscription != nil {
-				result = append(result, subscription.SubscriptionID)
+			if subscription != nil && subscription.SubscriptionID != nil {
+				result[*subscription.SubscriptionID] = strings.Dereference(subscription.DisplayName)
```

Comment:
Can this lead to a panic?

Reply:
No, I check the `subscription.SubscriptionID` for `nil`ness in the `if` above